c++/src/kj/mutex-test.c++

```diff
@@ -59,6 +59,11 @@ TEST(Mutex, MutexGuarded) {
     EXPECT_EQ(123u, *lock);
     EXPECT_EQ(123u, value.getAlreadyLockedExclusive());
 
+#if KJ_USE_FUTEX
+    EXPECT_TRUE(value.lockExclusiveWithTimeout(MILLISECONDS * 50) == nullptr);
```

Comment:
We should probably expand these tests a bit:

* Verify that, after a timeout, the expected amount of time has actually passed (wall time clock has advanced by at least the duration indicated).
* For the cases where the lock is acquired, actually do something with the lock.
* Maybe try to test for the bug where if two threads tried to take exclusive locks at the same time and one thread timed out, the other thread's exclusive lock would never complete due to having unset the flag?
* I guess the other bug (failing to send wakeup after reader timeout) is probably too hard to test (since the timing window is so narrow), so you can skip that.